Add Support to Minio DB

jsiloto opened the pull request.

Commit:
Add support for minio with file upload

Commit:
Add get file and metadata

Commit:
Add GET /image/

Commit:
Code cleanup

Commit:
Fix endpoints and bucket naming

mmagr commented on ImageManager/ImageManager.py:
```diff
@@ -40,10 +52,25 @@ def get_image(imageid):
+        init_tenant_context(request, db, minioClient)
+        orm_image = assert_image_exists(imageid)
+        filename = imageid + '.hex'
+        return send_from_directory(directory='/tmp/', filename=filename)
```
Here we're actually returning the "cache" copy used when the user uploads to minio?

jsiloto replied:
Fixed

mmagr commented on ImageManager/ImageManager.py:
```diff
@@ -94,21 +119,29 @@ def create_image():
-
+        extension = file_data.filename.rsplit('.', 1)[1].lower()
+        filename = imageid + '.' + extension
+        file_data.save(os.path.join('/tmp/', filename))
```
is this what we're serving through `GET /images/:id/binary`?

Commit:
Fix object retrieval from DB

Commit:
Fix problematic versioning of Marshmallow

requirements accepted versions >=3.0.0b3 but API changed on 3.0.0b7, see:
http://marshmallow.readthedocs.io/en/latest/api_reference.html#marshmallow.Schema.load

giovannicuriel commented on ImageManager/ImageManager.py:
```diff
@@ -1,35 +1,50 @@
+    try:
+        init_tenant_context(request, db, minioClient)
+        images = get_all_images()
+        json_images = [image_schema.dump(i).data for i in images]
```
I had a few problems with dump(x).data with the new Marshmallow version - apparently, dump() returns the dict directly (without `data` attribute)

jsiloto replied:
Yeah, I missed that.
Fixed

jsiloto committed:
Fix Marshmallow serialization
